src/qengine/operators.cpp

```diff
@@ -931,6 +931,62 @@ void QEngineCPU::DIV(bitCapInt toDiv, bitLenInt inOutStart, bitLenInt carryStart
     }
 }
 
+void QEngineCPU::MULModNOut(bitCapInt toMul, bitLenInt inStart, bitLenInt outStart, bitLenInt length)
```

Comment:
With the exception of lowMask and outRes, these two functions are almost identical.  Can they be merged together to reduce the overall code?

Ditto for the almost-identical controlled variants below.

Reply:
Do you see the major performance problem with branching in the lambda expression, though? I'm hesitant to do that.

Reply:
Alternative would be to branch outside the `par_for`? Except, that doesn't address your issue, and we'd give up the edge case logic up top.

Reply:
I could use a macro function (but not like last time). Let me give it a shot.

Reply:
How about just passing in a lambda?

Reply:
I did so. See if it fits the bill.

Reply:
Okay, I didn't see your comment. What I meant was, I used macros. If you strongly prefer a lambda within the lambda, I'm not sure how it will affect performance, but let me know.

Reply:
I'm writing a lambda version, but they are not appropriate for the OpenCL kernels, on a related note.